### PR TITLE
vmqemu package

### PR DESCRIPTION
**Pull request by tomponline**
- Moves VMQemu into own vmqemu package.
- Moves instance load and instance config validation functions into instance package (so usable from outside main package).
- Restructures backup package to not depend on instance package for loading by name (to prevent circular imports).

### CODE REVIEW

File: lxd/networks.go [hunk 1]
```diff
@@ -755,12 +755,14 @@ func networkLeasesGet(d *Daemon, r *http.Request) response.Response {
 					if err != nil {
 						continue
 					}
-				} else if inst.Type() == instancetype.VM {
-					d, err = inst.(*vmQemu).fillNetworkDevice(k, d)
+				}
+				/* tomp TODO why hasn't this been done at instance/device create time?
```
Comment (tomponline) [inline lxd/networks.go]: @stgraber I wanted to ask you about this part, I didn't want to add `fillNetworkDevice` as an exported function in the `Instance` interface, and wondered why we need to call `fillNetworkDevice` here, as it shouldnt the MAC address be generated when the device or instance is created?

File: lxd/instance/vmqemu/vm_qemu.go [hunk 2]
```diff
@@ -2898,46 +2954,56 @@ func (vm *vmQemu) ExpiryDate() time.Time {
...
+func (vm *VMQemu) LogPath() string {
 	name := project.Prefix(vm.Project(), vm.Name())
 	return shared.LogPath(name)
 }
 
-func (vm *vmQemu) LogFilePath() string {
-	return filepath.Join(vm.LogPath(), "lxvm.log")
+// LogFilePath returns the instance's log path.
+func (vm *VMQemu) LogFilePath() string {
+	return filepath.Join(vm.LogPath(), "lxvm.log") // tomp TODO is this correct?
```
Comment (tomponline) [inline lxd/instance/vmqemu/vm_qemu.go]: @stgraber I noticed that this file name doesn't look quite right. Should it be lxdvm.log or just vm.log?

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/instance/vmqemu/vm_qemu.go (hunk 2 above)
Comment (stgraber) [reply lxd/instance/vmqemu/vm_qemu.go]: I suspect we'll want `qemu.log` but this is unused so not that important.

File: lxd/networks.go (hunk 1 above)
Comment (stgraber) [reply lxd/networks.go]: `fillNetworkDevice` is still needed to populate `d["hwaddr"]` from the relevant volatile key.

**Comment by stgraber** [comment]
The package name should be just `qemu`, `instance/qemu` is specific enough as far as import path and we can't do containers with qemu or VMs with lxc, so having `instance/qemu` and `instance/lxc` is probably best.

The struct itself would become `Qemu` then.

**Comment by stgraber** [comment]
Keeping this PR open for now as we need this and the network dhcp lease code fixed.

**Comment by stgraber** [comment]
Fixed the pull request, will merge if Jenkins looks happy with it.

**Pull request merged**